quint/src/types/constraintGenerator.ts

```diff
@@ -166,6 +167,33 @@ export class ConstraintGeneratorVisitor implements IRVisitor {
     this.addToResults(e.id, right(toScheme({ kind: e.kind })))
   }
 
+  exitTuple(e: QuintTup) {
```

Comment:
As we did with the effects inferrer, I think we can also generalize something here so we don't have to repeat this code between `exitTuple` and `exitApp`. This will also repeat with the other structures (lists, maps, records, etc), so we can try to generalize when adding the next one.